avocado/utils/cpu.py

```diff
@@ -356,3 +358,28 @@ def get_pid_cpus(pid):
         except IOError:
             continue
     return list(cpus)
+
+
+def is_smt():
+
+    """
+    Return True/False Based on if system has hyperthreading enabled
+    :return: True/False
+    :rtype: boolean
+    """
+
+    if get_cpu_vendor_name().startswith('power'):
+        try:
+            if b'is not SMT capable' in process.system_output("ppc64_cpu --smt"):
```

Comment:
Based on the powerpc-utils source code, it looks the check for smt can be done simply by looking at `/sys`.  Take a look at https://github.com/ibm-power-utilities/powerpc-utils/blob/a33f99be688c42f5eeefea2dfd3c69a908f489e2/src/ppc64_cpu.c#L374 

```c
static int is_smt_capable(void)
{
	struct stat sb;
	char path[SYSFS_PATH_MAX];
	int i;

	for (i = 0; i < threads_in_system; i++) {
		sprintf(path, SYSFS_CPUDIR"/smt_snooze_delay", i);
		if (stat(path, &sb))
			continue;
		return 1;
	}

	return 0;
}


```

Which can be loosely translated into Python as:

```
def is_smt_capable():
   for cpu_dir in glob.glob("/sys/devices/system/cpu/cpu[0-9]*"):
      if os.path.exists(os.path.join(cpu_dir, 'smt_snooze_delay')):
         return True
   return False
```

I've tested this on a Power 9 machine, and it seems to work as intended.  I believe you have wider access to different machines, so please test that.

Reply:
@clebergnu thanks a lot on helping this, let me adopt and test in my environment